File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i in range(len(nums)):
            val = nums[i]
            hash[val] = i
        

        for i in range(len(nums)):
            val = nums[i]

            if target - val in hash and hash[target-val] != i:
                return [i,hash[target-val]]

        return [-1,-1]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.go
func twoSum(numbers []int, target int) []int {
    i := 0
    j := len(numbers)-1

    for i<j {
        if numbers[i] + numbers[j] > target {
            j = j -1
        }else if numbers[i] + numbers[j] < target {
            i = i + 1
        }else {
            return []int{i+1,j+1}
        }
    }
    return []int{0,0}
}
